plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/BaseSystemTable.java

```diff
@@ -98,12 +99,26 @@ private List<Page> buildPages(ConnectorTableMetadata tableMetadata, ConnectorSes
     private void addRows(DataTask dataTask, PageListBuilder pagesBuilder, TimeZoneKey timeZoneKey, Map<String, Integer> columnNameToPositionInSchema)
     {
         try (CloseableIterable<StructLike> dataRows = dataTask.rows()) {
-            dataRows.forEach(dataTaskRow -> addRow(pagesBuilder, dataTaskRow, timeZoneKey, columnNameToPositionInSchema));
+            dataRows.forEach(dataTaskRow -> addRow(pagesBuilder, new Row(dataTaskRow, columnNameToPositionInSchema), timeZoneKey));
         }
         catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
-    protected abstract void addRow(PageListBuilder pagesBuilder, StructLike structLike, TimeZoneKey timeZoneKey, Map<String, Integer> columnNameToPositionInSchema);
+    protected abstract void addRow(PageListBuilder pagesBuilder, Row row, TimeZoneKey timeZoneKey);
+
+    public record Row(StructLike structLike, Map<String, Integer> columnNameToPositionInSchema)
+    {
+        public Row
+        {
+            requireNonNull(structLike, "structLike is null");
+            columnNameToPositionInSchema = ImmutableMap.copyOf(requireNonNull(columnNameToPositionInSchema, "columnNameToPositionInSchema is null"));
```

Comment:
nit: can skip `requireNonNull` here, as attempt to copy null will trigger exception anyway